hack/update-kyverno-crds.sh

```diff
@@ -29,36 +29,117 @@ EOF
+crds=$(mktemp)
+curl -sL "$url" | yq eval-all 'select(.kind == "CustomResourceDefinition")' - > "$crds"
+
+# Split on "---" manually
+awk -v outdir="$crd_dir" -v version="$version" '
+  BEGIN {
+    fileIndex = 0;
+    crd = "";
+  }
+  /^---/ {
+    if (crd != "") {
+      individual_crd_tempfile = "/tmp/tmp-crd-" fileIndex ".yaml"
+      print crd > individual_crd_tempfile;
+
+      cmd_get_plural = "yq eval \".spec.names.plural\" " individual_crd_tempfile
+      pluralName = "unknown-plural-" fileIndex
+      if ((cmd_get_plural | getline pn_line) > 0 && pn_line != "" && pn_line != "null") {
+        pluralName = pn_line
+      }
+      close(cmd_get_plural)
+
+      cmd_get_group = "yq eval \".spec.group\" " individual_crd_tempfile
+      groupName = ""
+      if ((cmd_get_group | getline gn_line) > 0 && gn_line != "" && gn_line != "null") {
+        groupName = gn_line
+      }
+      close(cmd_get_group)
+      
+      if (groupName == "") {
+        final_filename = outdir "/" pluralName ".yaml"
+      } else {
+        final_filename = outdir "/" pluralName "." groupName ".yaml"
+      }
+
+      cmd_process_crd = "yq eval '\''del(.metadata.creationTimestamp)'\'' " individual_crd_tempfile
+      print "# This file has been generated by hack/update-kyverno-crds.sh using Kyverno version: " version ". Do not edit." > final_filename
+      while ((cmd_process_crd | getline proc_line) > 0) {
+        print proc_line >> final_filename
+      }
+      close(cmd_process_crd)
+
+      system("rm -f " individual_crd_tempfile)
+      crd = "";
+      fileIndex++;
+    }
+    next;
+  }
+  {
+    crd = crd $0 "\n";
+  }
+  END {
+    if (crd != "") {
+      individual_crd_tempfile = "/tmp/tmp-crd-" fileIndex ".yaml"
+      print crd > individual_crd_tempfile;
+
+      cmd_get_plural = "yq eval \".spec.names.plural\" " individual_crd_tempfile
+      pluralName = "unknown-plural-" fileIndex
+      if ((cmd_get_plural | getline pn_line) > 0 && pn_line != "" && pn_line != "null") {
+        pluralName = pn_line
+      }
+      close(cmd_get_plural)
+
+      cmd_get_group = "yq eval \".spec.group\" " individual_crd_tempfile
+      groupName = ""
+      if ((cmd_get_group | getline gn_line) > 0 && gn_line != "" && gn_line != "null") {
+        groupName = gn_line
+      }
+      close(cmd_get_group)
+      
+      if (groupName == "") {
+        final_filename = outdir "/" pluralName ".yaml"
+      } else {
+        final_filename = outdir "/" pluralName "." groupName ".yaml"
+      }
+
+      cmd_process_crd = "yq eval '\''del(.metadata.creationTimestamp)'\'' " individual_crd_tempfile
+      print "# This file has been generated by hack/update-kyverno-crds.sh with Kyverno version " version ". Do not edit." > final_filename
+      while ((cmd_process_crd | getline proc_line) > 0) {
+        print proc_line >> final_filename
+      }
+      close(cmd_process_crd)
+
+      system("rm -f " individual_crd_tempfile)
+    }
+  }
```

Comment:
Frankly saying: `yq e 'del(.metadata.creationTimestamp)'` looked much better.

Reply:
True, but the problem was that Kyverno has two different CRDs with the same plural name, and when running it like we were running we were missing registering one CRD in our UCs, this was a bug from the script. And it had another problem with the comment, the files didn't have any comment at all, the `while` was not being executed.

Reply:
no, this is too terrible, let's use another split 
```
yq eval-all --split-exp .metadata.name 'select(.kind == "CustomResourceDefinition")'
```

Reply:
I updated the script, we split by name but also have to reprocess each file. Also `.metadata.name` would have to be `'(.metadata.name + ".yaml")'` in order for it to work. WDYT?